### PR TITLE
Deprecate support for out-dated platforms in CuPy v13

### LINKED ISSUE #6805: Change rapidsai CI image to stable once pylibraft support is released
Due to #6690 changing the `rapidsai/rapidsai-core-dev-nightly` image to nightly, we should pin it to a stable one once all the required features are released.

### LINKED ISSUE #7557: [RFC] Drop support for CUDA 10.2/11.0/11.1 in CuPy v13
The CuPy team plans to drop support for CUDA 10.2, 11.0, and 11.1 in CuPy v13, which will be released in October 2023 (#7555). The primary motivation for this change is to reduce the work necessary to publish a CuPy release. These legacy CUDA versions require separate packaging for each version as they don't support CUDA Minor Compatibility policy. Reducing the number of packages also helps keep the PyPI community healthy in terms of the disk space and bandwidth required to distribute packages.

Here are the download counts of our binary packages in the recent three months (sdist and `cupy-wheel` are excluded):

CuPy PyPI Binary Package Downloads (Feb 2023 - Apr 2023)

According to the statistics, 22.3% of users will be affected by this change. However, users of these CUDA versions can continue to use CuPy v12, which we believe is stable enough and will continue to be maintained until the release of the CuPy v13. The number also shows that 31.2% of users are still on CUDA 8.0~10.1, which is no longer supported, indicating they are satisfied with the legacy CuPy releases (v9 or earlier). That said, we recommend upgrading to the latest CUDA release, which provides better performance.

Please leave a comment on this issue if you have any feedback on this change.

### PR DESCRIPTION
**Pull request by kmaehashi**
This PR drops support for:

* CUDA 11.1 or earlier (closes #7557)
* Python 3.8 (#7555)
* NumPy 1.21 (NEP 29)
* cuTENSOR 1.5 or earlier
* NCCL 2.15 or earlier
* cuDNN 8.8 or earlier
* Ubuntu 18.04 (#7127)

Also closes #6805

### CODE REVIEW

**Comment by kmaehashi** [comment]
Updated `/ test` tags (thansks @asi1024 for pointing this out!)
`mini` now triggers CI for CUDA {11.2, 11.8, 12.0, 12.1} x {Linux, Windows}.

File: .pfnci/linux/tests/cuda-rapids.Dockerfile [hunk 1]
```diff
@@ -1,5 +1,4 @@
-# TODO(kmaehashi): Fix stable image after release (#6805)
-ARG BASE_IMAGE="rapidsai/rapidsai-core-dev-nightly:22.08-cuda11.5-devel-ubuntu20.04-py3.8"
+ARG BASE_IMAGE="rapidsai/rapidsai-core-dev:23.06-cuda11.8-devel-ubuntu22.04-py3.10"
```
Comment (emcastillo) [inline .pfnci/linux/tests/cuda-rapids.Dockerfile:1]: I think that this image is not working well :(

Comment (emcastillo) [reply .pfnci/linux/tests/cuda-rapids.Dockerfile:1]: https://github.com/rapidsai/cugraph/pull/2866

This PR sseems to have changed the API of `weakly_connected_components`

Comment (kmaehashi) [inline .pfnci/linux/tests/cuda-rapids.Dockerfile:1]: Thanks! Let's pin to the old one and fix that in a separate pull-request.

```suggestion
ARG BASE_IMAGE="rapidsai/rapidsai-core-dev:22.10-cuda11.5-devel-ubuntu20.04-py3.9"
```

**Pull request merged**